README.md

```diff
@@ -1 +1,329 @@
+
+> [Implementers] may provide an opt-in for systems where round-trip determinism is a desireable [sic] feature and backward compatibility with old, non-strict data is unnecessary.
+>
+> — [DAG-JSON Spec][DAG-JSON]
+
+As it is critical for guarding against various attacks, the assumptions around canonical encoding MUST be enforced.
+
+## 2.1 Signing CIDs
+
+Rather than validating the inline IPLD, replacing the data with a CID link to the content MAY be used instead. Note while this is very safe (as it impractical to alter a signed hash), this approach mixes data layout with security, and may have a performance, disk, and networking impacts.
+
+### 2.1.1 Caching & Invalidation
+
+Signing CIDs has two additional caching effects
+
+* Signing CIDs enables a simple strategy for caching validation by CID
+* Such a strategy also requires accounting for revocation of the signing keys, and so need to be tagged with this additional metadata
+
+## 2.2 Raw (Noncanonicalized) Data
+
+Canonicalization is not required if data is encoded as raw bytes (multicodec `0x55`). The exact bytes are already present, and MUST not be changed.
+
+# 3 Varsig Format
+
+After being decoded from [unsigned varint]s, a varsig includes the following segments:
+
+```abnf
+varsig = multibase-prefix %x34 varsig-header varsig-body
+multibase-prefix = ALPHA ; Multibase
+varsig-header = unsigned-varint ; Usually the public key code from Multicodec
+varsig-body = *unsigned-varint; Zero or more segments required by the kind of varsig (e.g. raw bytes, hash algorithm, etc)
+```
+
+For example, here is an EdDSA signature for some content encoded as DAG-PB:
+
+`0x34ed01ae3784f03f9ee1163382fa6efa73b0c31ecf58c899c836709303ba4621d1e6df20e09aaa568914290b7ea124f5b38e70b9b69c7de0d216880eac885edd41c302`
+
+### 3.1 Varsig Prefix
+
+The varsig prefix MUST be `0x34`.
+
+### 3.2 Signature Header
+
+The prefix of the signature algorithm. This is often the [multicodec] of the associated public key, but MAY be unique for the signature type. The code MAY live outside the multicodec table. This field MUST act as a discriminant for how many expected fields come in the varsig body, and what each of them mean.
+
+### 3.3 Varsig Body
+
+The varsig body MUST consist of zero or more segments, and MUST be defined by the signature algorithm.
+
+Some examples include:
+
+* Raw signature bytes only
+* CID of [DKIM] certification transparency record, and raw signature bytes
+* Hash algorithm multicodec prefix, signature counter, HMAC, and raw signature bytes
+
+# 4 Registry of Common Signature Algorithms
+
+Below are a few common signature headers and their fields.
+
+## 4.1 RSA
+
+RSASSA-PKCS #1 v1.5 signatures MUST include the following segments:
+
+``` abnf
+rsa-varsig-header = 0x1205 ; RSASSA-PKCS #1 v1.5
+rsa-hash-algorithm = 2*OCTET
+sig-bytes = varint
+```
+
+### 4.1.1 Example: RS256
+
+``` abnf
+rsa-varsig-header = 0x1205 ; RSASSA-PKCS #1 v1.5 multicodec prefix
+rsa-hash-algorithm = 0x12 ; From multicodec
```

Comment:
`0x12` is actually a varint as well right? Just that `0x12 = varint(0x12)`?

Reply:
That's a good point, we shoud be clear about what these are (e.g. `0xed` vs `0xed01`). I'm pretty sure that the multicodec table gives values _not_ in varint format, and you have to varint it yourself, but I should tripple check 👀